azurerm/internal/services/appconfiguration/tests/app_configuration_resource_test.go

```diff
@@ -90,6 +90,60 @@ func TestAccAppConfigurationResource_complete(t *testing.T) {
 	})
 }
 
+func TestAccAppConfigurationResource_identity(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_app_configuration", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAppConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAppConfigurationResource_identity(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAppConfigurationExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAppConfigurationResource_identityUpdated(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_app_configuration", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAppConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAppConfigurationResource_standard(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAppConfigurationExists(data.ResourceName),
+				),
+			},
+			{
+				Config: testAppConfigurationResource_identity(data),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "identity.#", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "identity.0.type", "SystemAssigned"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "identity.0.principal_id"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "identity.0.tenant_id"),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAppConfigurationResource_standard(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAppConfigurationExists(data.ResourceName),
+				),
+				ExpectNonEmptyPlan: true,
```

Comment:
why is this a non-empty plan?

Reply:
I thought it's supposed to be non-empty, cause it removed the identity here?

Reply:
If I remove it, I get a non-empty plan, but I am not sure what I am doing wrong?

```
 $ TF_ACC=1 go test -v ./azurerm/internal/services/appconfiguration/tests -timeout=600m -run=TestAccAppConfigurationResource_identityUpdated
=== RUN   TestAccAppConfigurationResource_identityUpdated
=== PAUSE TestAccAppConfigurationResource_identityUpdated
=== CONT  TestAccAppConfigurationResource_identityUpdated
    testing.go:684: Step 3 error: After applying this step, the plan was not empty:

        DIFF:

        UPDATE: azurerm_app_configuration.test
          endpoint:                                "https://testaccappconf201019100005359001.azconfig.io" => "https://testaccappconf201019100005359001.azconfig.io"
          id:                                      "/subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestrg-201019100005359001/providers/Microsoft.AppConfiguration/configurationStores/testaccappconf201019100005359001" => "/subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestrg-201019100005359001/providers/Microsoft.AppConfiguration/configurationStores/testaccappconf201019100005359001"
          identity.#:                              "1" => "0"
          identity.0.principal_id:                 "c8c17002-52eb-429c-91f8-22b631892025" => ""
          identity.0.tenant_id:                    "b945c813-dce6-41f8-8457-5a12c2fe15bf" => ""
          identity.0.type:                         "SystemAssigned" => ""
          location:                                "eastus2" => "eastus2"
          name:                                    "testaccappconf201019100005359001" => "testaccappconf201019100005359001"
          primary_read_key.#:                      "1" => "1"
          primary_read_key.0.connection_string:    "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=KTnC-l6-s0:brDVit1hBYc6B+aIKqyr;Secret=hMR//68axkn5HF4URMcTzGDcxuaqHKQYP0XQTxHEEWQ=" => "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=KTnC-l6-s0:brDVit1hBYc6B+aIKqyr;Secret=hMR//68axkn5HF4URMcTzGDcxuaqHKQYP0XQTxHEEWQ="
          primary_read_key.0.id:                   "KTnC-l6-s0:brDVit1hBYc6B+aIKqyr" => "KTnC-l6-s0:brDVit1hBYc6B+aIKqyr"
          primary_read_key.0.secret:               "hMR//68axkn5HF4URMcTzGDcxuaqHKQYP0XQTxHEEWQ=" => "hMR//68axkn5HF4URMcTzGDcxuaqHKQYP0XQTxHEEWQ="
          primary_write_key.#:                     "1" => "1"
          primary_write_key.0.connection_string:   "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=b2eJ-l6-s0:EFAlqcI/fLBPV3x1kb9w;Secret=ZypBiRzopT/oRjuHDYbTsTIS3dLxIpTkzhugfkGrfys=" => "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=b2eJ-l6-s0:EFAlqcI/fLBPV3x1kb9w;Secret=ZypBiRzopT/oRjuHDYbTsTIS3dLxIpTkzhugfkGrfys="
          primary_write_key.0.id:                  "b2eJ-l6-s0:EFAlqcI/fLBPV3x1kb9w" => "b2eJ-l6-s0:EFAlqcI/fLBPV3x1kb9w"
          primary_write_key.0.secret:              "ZypBiRzopT/oRjuHDYbTsTIS3dLxIpTkzhugfkGrfys=" => "ZypBiRzopT/oRjuHDYbTsTIS3dLxIpTkzhugfkGrfys="
          resource_group_name:                     "acctestrg-201019100005359001" => "acctestrg-201019100005359001"
          secondary_read_key.#:                    "1" => "1"
          secondary_read_key.0.connection_string:  "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=/408-l6-s0:eHVrHaM9Wu2eVtMWNsPl;Secret=xLcLbhpIphpTIrL3Ve8BjQo/Eml2W88yeyHRRHeI+rA=" => "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=/408-l6-s0:eHVrHaM9Wu2eVtMWNsPl;Secret=xLcLbhpIphpTIrL3Ve8BjQo/Eml2W88yeyHRRHeI+rA="
          secondary_read_key.0.id:                 "/408-l6-s0:eHVrHaM9Wu2eVtMWNsPl" => "/408-l6-s0:eHVrHaM9Wu2eVtMWNsPl"
          secondary_read_key.0.secret:             "xLcLbhpIphpTIrL3Ve8BjQo/Eml2W88yeyHRRHeI+rA=" => "xLcLbhpIphpTIrL3Ve8BjQo/Eml2W88yeyHRRHeI+rA="
          secondary_write_key.#:                   "1" => "1"
          secondary_write_key.0.connection_string: "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=Y/am-l6-s0:8/AtQs6woCBeLOXx2fSh;Secret=o77bgostJoKDLl4M9POs4L3xNspnDReR3hH0J3AbQ1Y=" => "Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=Y/am-l6-s0:8/AtQs6woCBeLOXx2fSh;Secret=o77bgostJoKDLl4M9POs4L3xNspnDReR3hH0J3AbQ1Y="
          secondary_write_key.0.id:                "Y/am-l6-s0:8/AtQs6woCBeLOXx2fSh" => "Y/am-l6-s0:8/AtQs6woCBeLOXx2fSh"
          secondary_write_key.0.secret:            "o77bgostJoKDLl4M9POs4L3xNspnDReR3hH0J3AbQ1Y=" => "o77bgostJoKDLl4M9POs4L3xNspnDReR3hH0J3AbQ1Y="
          sku:                                     "standard" => "standard"



        STATE:

        azurerm_app_configuration.test:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestrg-201019100005359001/providers/Microsoft.AppConfiguration/configurationStores/testaccappconf201019100005359001
          provider = provider.azurerm
          endpoint = https://testaccappconf201019100005359001.azconfig.io
          identity.# = 1
          identity.0.principal_id = c8c17002-52eb-429c-91f8-22b631892025
          identity.0.tenant_id = b945c813-dce6-41f8-8457-5a12c2fe15bf
          identity.0.type = SystemAssigned
          location = eastus2
          name = testaccappconf201019100005359001
          primary_read_key.# = 1
          primary_read_key.0.connection_string = Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=KTnC-l6-s0:brDVit1hBYc6B+aIKqyr;Secret=hMR//68axkn5HF4URMcTzGDcxuaqHKQYP0XQTxHEEWQ=
          primary_read_key.0.id = KTnC-l6-s0:brDVit1hBYc6B+aIKqyr
          primary_read_key.0.secret = hMR//68axkn5HF4URMcTzGDcxuaqHKQYP0XQTxHEEWQ=
          primary_write_key.# = 1
          primary_write_key.0.connection_string = Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=b2eJ-l6-s0:EFAlqcI/fLBPV3x1kb9w;Secret=ZypBiRzopT/oRjuHDYbTsTIS3dLxIpTkzhugfkGrfys=
          primary_write_key.0.id = b2eJ-l6-s0:EFAlqcI/fLBPV3x1kb9w
          primary_write_key.0.secret = ZypBiRzopT/oRjuHDYbTsTIS3dLxIpTkzhugfkGrfys=
          resource_group_name = acctestrg-201019100005359001
          secondary_read_key.# = 1
          secondary_read_key.0.connection_string = Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=/408-l6-s0:eHVrHaM9Wu2eVtMWNsPl;Secret=xLcLbhpIphpTIrL3Ve8BjQo/Eml2W88yeyHRRHeI+rA=
          secondary_read_key.0.id = /408-l6-s0:eHVrHaM9Wu2eVtMWNsPl
          secondary_read_key.0.secret = xLcLbhpIphpTIrL3Ve8BjQo/Eml2W88yeyHRRHeI+rA=
          secondary_write_key.# = 1
          secondary_write_key.0.connection_string = Endpoint=https://testaccappconf201019100005359001.azconfig.io;Id=Y/am-l6-s0:8/AtQs6woCBeLOXx2fSh;Secret=o77bgostJoKDLl4M9POs4L3xNspnDReR3hH0J3AbQ1Y=
          secondary_write_key.0.id = Y/am-l6-s0:8/AtQs6woCBeLOXx2fSh
          secondary_write_key.0.secret = o77bgostJoKDLl4M9POs4L3xNspnDReR3hH0J3AbQ1Y=
          sku = standard

          Dependencies:
            azurerm_resource_group.test
        azurerm_resource_group.test:
          ID = /subscriptions/a31346ad-8afd-4adb-910b-4865f7f6ca0f/resourceGroups/acctestRG-201019100005359001
          provider = provider.azurerm
          location = eastus2
          name = acctestRG-201019100005359001
```